Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml

```diff
@@ -6605,6 +6743,11 @@
             <Button DefaultButtonFace="AP_SwarmHost" Requirements="AP_HaveInfestationPit"/>
             <Unit value="AP_SwarmHost"/>
         </InfoArray>
+        <InfoArray index="Train16" Time="10">
+            <!-- TODO: Add "opt-in" upgrade and have this require that instead of lair-->
+            <Button DefaultButtonFace="AP_Morphling" Requirements="AP_HaveLair"/>

```

Comment:
Are you planning to have the upgrade replace lair, or be in addition to it?

Reply:
Not sure, honestly. 

Pros of requiring Lair:
- You start with a Lair on every mission
- The only adv morph that doesn't _also_ require a Lair is the Baneling
- Losing your Lair should be reasonably punishing
- It feels weird having literally no internal requirement for this unit (ie. the only other requirement can only be granted by an external-facing trigger)

Cons of requiring a Lair:
- If literally all you have is Banelings you need all the help you can get
- 99% of the time nobody is going to notice the requirement anyway so who cares

Alternatives:
- It could require the upgrade, AND ALSO a combination of the tech unlock + structure for the unit in question... but that would look really messy on the requirements panel, and I don't know if it's actually possible to check in requirement nodes whether tech is available for a unit

...Can probably just leave it off for now. But since I'm making changes anyway I'll just throw the upgrade in I think. Adding the unit to AP_Triggers will need to be a different PR anyway, so I'll hook up the "grant upgrade by default" trigger there too.